File: plugin/js/framework/next.js/v1_x/methods.js
function detect(input, callback) {
  let path = input.request.root;
  callback(null, { value: true, error: null });
  // callback(reject, { value: null, error: reject })
}

function isFrameworkUsed(input, callback) {
  let path = input.request.root;
  callback(null, { value: true, error: null });
  // callback(reject, { value: null, error: reject })
}

function percentOfFrameworkUsed(input, callback) {
  let path = input.request.root;
  callback(null, { error: null });
  // callback(reject, { value: null, error: reject })
}

module.exports = {
  detect: detect,
  isFrameworkUsed: isFrameworkUsed,
  percentOfFrameworkUsed: percentOfFrameworkUsed,
};

File: plugin/python/framework/django/v1_x/methods.py
from protos.plugin import framework_pb2_grpc, common_pb2


class Methods(framework_pb2_grpc.FrameworkServiceServicer):
    def Detect(self, request, context):
        return common_pb2.ServiceOutputBool(value=True, error=None)

    def IsFrameworkUsed(self, request, context):
        return common_pb2.ServiceOutputBool(value=True, error=None)

    def PercentOfFrameworkUsed(self, request, context):
        return common_pb2.ServiceOutputFloat(value=78, error=None)
